code/game/machinery/constructable_frame.dm

```diff
@@ -223,61 +225,58 @@ Note: Once everything is added to the public areas, will add MAT_METAL and MAT_G
-		/obj/machinery/vending/autodrobe = "AutoDrobe",
-		/obj/machinery/vending/clothing = "ClothesMate",
-		/obj/machinery/vending/medical = "NanoMed Plus",
-		/obj/machinery/vending/wallmed = "NanoMed",
-		/obj/machinery/vending/assist  = "Vendomat",
-		/obj/machinery/vending/engivend = "Engi-Vend",
-		/obj/machinery/vending/hydronutrients = "NutriMax",
-		/obj/machinery/vending/hydroseeds = "MegaSeed Servitor",
-		/obj/machinery/vending/sustenance = "Sustenance Vendor",
-		/obj/machinery/vending/dinnerware = "Plasteel Chef's Dinnerware Vendor",
-		/obj/machinery/vending/cart = "PTech",
-		/obj/machinery/vending/robotics = "Robotech Deluxe",
-		/obj/machinery/vending/engineering = "Robco Tool Maker",
-		/obj/machinery/vending/sovietsoda = "BODA",
-		/obj/machinery/vending/security = "SecTech",
-		/obj/machinery/vending/crittercare = "CritterCare")
+		"Booze-O-Mat" =							/obj/machinery/vending/boozeomat,
+		"Solar's Best Hot Drinks" =				/obj/machinery/vending/coffee,
+		"Getmore Chocolate Corp" =				/obj/machinery/vending/snack,
+		"Mr. Chang" =							/obj/machinery/vending/chinese,
+		"Robust Softdrinks" =					/obj/machinery/vending/cola,
+		"ShadyCigs Deluxe" =					/obj/machinery/vending/cigarette,
+		"Hatlord 9000" =						/obj/machinery/vending/hatdispenser,
+		"Suitlord 9000" =						/obj/machinery/vending/suitdispenser,
+		"Shoelord 9000" =						/obj/machinery/vending/shoedispenser,
+		"AutoDrobe" =							/obj/machinery/vending/autodrobe,
+		"ClothesMate" =							/obj/machinery/vending/clothing,
+		"NanoMed Plus" =						/obj/machinery/vending/medical,
+		"NanoMed" =								/obj/machinery/vending/wallmed,
+		"Vendomat" =							/obj/machinery/vending/assist,
+		"Engi-Vend" =							/obj/machinery/vending/engivend,
+		"NutriMax" =							/obj/machinery/vending/hydronutrients,
+		"MegaSeed Servitor" =					/obj/machinery/vending/hydroseeds,
+		"Sustenance Vendor" =					/obj/machinery/vending/sustenance,
+		"Plasteel Chef's Dinnerware Vendor" =	/obj/machinery/vending/dinnerware,
+		"PTech" =								/obj/machinery/vending/cart,
+		"Robotech Deluxe" =						/obj/machinery/vending/robotics,
+		"Robco Tool Maker" =					/obj/machinery/vending/engineering,
+		"BODA" =								/obj/machinery/vending/sovietsoda,
+		"SecTech" =								/obj/machinery/vending/security,
+		"CritterCare" =							/obj/machinery/vending/crittercare)
 
 /obj/item/circuitboard/vendor/screwdriver_act(mob/user, obj/item/I)
 	. = TRUE
 	if(!I.use_tool(src, user, 0, volume = I.tool_volume))
 		return
-	var/static/list/display_vending_names_paths
-	if(!display_vending_names_paths)
-		display_vending_names_paths = list()
-		for(var/path in vending_names_paths)
-			display_vending_names_paths[vending_names_paths[path]] = path
-	var/choice =  input(user, "Choose a new brand","Select an Item") as null|anything in display_vending_names_paths
+	var/choice =  input(user, "Choose a new brand", "Select an Item") as null|anything in vending_names_paths
```

Comment:
It's possible for `choice` to equal null if the user chooses cancel, which gets passed to `set_type` and results in a runtime.